javascriptv3/example_code/cross-services/wkflw-resilient-service/steps-deploy.js

```diff
@@ -0,0 +1,623 @@
+  }),
+  new ScenarioOutput("createdLoadBalancer", (c) =>
+    MESSAGES.createdLoadBalancer
+      .replace("${LB_NAME}", NAMES.loadBalancerName)
+      .replace("${DNS_NAME}", c.loadBalancerDns),
+  ),
+  new ScenarioOutput(
+    "creatingListener",
+    MESSAGES.creatingLoadBalancerListener
+      .replace("${LB_NAME}", NAMES.loadBalancerName)
+      .replace("${TARGET_GROUP_NAME}", NAMES.loadBalancerTargetGroupName),
+  ),
+  new ScenarioAction("createListener", async (c) => {
+    // snippet-start:[javascript.v3.wkflw.resilient.CreateListener]
+    const client = new ElasticLoadBalancingV2Client({});
+    const { Listeners } = await client.send(
+      new CreateListenerCommand({
+        LoadBalancerArn: c.loadBalancerArn,
+        Protocol: c.targetGroupProtocol,
+        Port: c.targetGroupPort,
+        DefaultActions: [{ Type: "forward", TargetGroupArn: c.targetGroupArn }],
+      }),
+    );
+    // snippet-end:[javascript.v3.wkflw.resilient.CreateListener]
+    const listener = Listeners[0];
+    c.lbListenerArn = listener.ListenerArn;
+  }),
+  new ScenarioOutput("createdListener", (c) =>
+    MESSAGES.createdLoadBalancerListener.replace(
+      "${LB_LISTENER_ARN}",
+      c.lbListenerArn,
+    ),
+  ),
+  new ScenarioOutput(
+    "attachingLoadBalancerTargetGroup",
+    MESSAGES.attachingLoadBalancerTargetGroup
+      .replace("${TARGET_GROUP_NAME}", NAMES.loadBalancerTargetGroupName)
+      .replace("${AUTO_SCALING_GROUP_NAME}", NAMES.autoScalingGroupName),
+  ),
+  new ScenarioAction("attachLoadBalancerTargetGroup", async (c) => {
+    // snippet-start:[javascript.v3.wkflw.resilient.AttachTargetGroup]
+    const client = new AutoScalingClient({});
+    await client.send(
+      new AttachLoadBalancerTargetGroupsCommand({
+        AutoScalingGroupName: NAMES.autoScalingGroupName,
+        TargetGroupARNs: [c.targetGroupArn],
+      }),
+    );
+    // snippet-end:[javascript.v3.wkflw.resilient.AttachTargetGroup]
+  }),
+  new ScenarioOutput(
+    "attachedLoadBalancerTargetGroup",
+    MESSAGES.attachedLoadBalancerTargetGroup,
+  ),
+  new ScenarioOutput("verifyingInboundPort", MESSAGES.verifyingInboundPort),
+  new ScenarioAction(
+    "verifyInboundPort",
+    /**
+     *
+     * @param {{ defaultSecurityGroup: import('@aws-sdk/client-ec2').SecurityGroup}} c
+     */
+    async (c) => {
+      const client = new EC2Client({});
+      const { SecurityGroups } = await client.send(
+        new DescribeSecurityGroupsCommand({
+          Filters: [{ Name: "group-name", Values: ["default"] }],
+        }),
+      );
+      if (!SecurityGroups) {
+        c.verifyInboundPortError = new Error(MESSAGES.noSecurityGroups);
+      }
+      c.defaultSecurityGroup = SecurityGroups[0];
+
+      /**
+       * @type {string}
+       */
+      const ipResponse = (await axios.get("http://checkip.amazonaws.com")).data;
+      c.myIp = ipResponse.trim();
+      const myIpRules = c.defaultSecurityGroup.IpPermissions.filter((p) =>
```

Comment:
Re Jason's comments on short variable names, destructuring would I think be very nice here.

Reply:
Fixed.